import static org.junit.Assert.assertEquals;
import java.io.IOException;

import org.junit.Test;
import junit.framework.*;
import static org.junit.Assert.*;

public class Testavatar {

	
  @Test
public void testCurrentRoom() throws IOException{
	Room r = new Room();
	r.setName("Room 5");
	Avatar a = new Avatar(); 
	a.setNewRoom(r); 
	assertNotNull(r); 
	assertNotNull(a); 
	assertEquals(a.currentRoom, r); 
	assertEquals(a.currentRoom, r); 
}
  
@Test 
public void testcouseunFinishedcourse(){
	String teacherFile = "../Javaparttwo/teacher.text";
	String bookFile ="../Javaparttwo/book.text";
	String roomFile = "../Javaparttwo/sr.text";
	World w;
	try {
		w = new World();
	} catch (IOException e) {
		// TODO Auto-generated catch block
		e.printStackTrace();
	}
	try {
		w = new World(); 
		w.createWorld(teacherFile, bookFile, roomFile);
	} catch (IOException e) {
		// TODO Auto-generated catch block
		e.printStackTrace();
	}
	Avatar a = new Avatar();
	assertNotNull(w); 
	assertNotNull(a);
	Course c1 = w.returnRandomCourse(); 
	Course c2 = w.returnRandomCourse(); 
	Course c3 = w.returnRandomCourse(); 
	Course c4 = w.returnRandomCourse(); 
	assertNotNull(c1); 
	assertNotNull(c2); 
	assertNotNull(c3); 
	assertNotNull(c4); 
	a.finishedCourse.add(c4);  // giving me null pointer exception 
	
}



	
	
}
